packages/engine/src/simulation/command.rs

```diff
@@ -33,23 +43,25 @@ struct RemoveCommand {
     uuid: Uuid,
 }
 
-/// TODO: DOC
 #[derive(Default)]
 pub struct CreateRemoveCommands {
     create: Vec<CreateCommand>,
     remove: Vec<RemoveCommand>,
 }
 
 impl CreateRemoveCommands {
+    /// Push a command for the request of the creation of an agent
     pub fn add_create(&mut self, agent: Agent) {
         self.create.push(CreateCommand { agent });
     }
 
+    /// Push a command for the request of the deletion of the agent associated with the given UUID
     pub fn add_remove(&mut self, uuid: Uuid) {
         self.remove.push(RemoveCommand { uuid });
     }
 
-    /// TODO: DOC
+    /// Ensures that all agent-creation commands contain valid agent fields. Returns an error if
+    /// a creation command is for an agent that has a field that hasn't been defined in the schema
```

Comment:
```suggestion
    /// Ensures that all agent-creation commands contain valid agent fields. 
    ///
    /// # Errors
    ///
    /// If a creation command is for an agent that has a field that hasn't been defined in the schema.
```

Reply:
This isn't a comprehensive list of errors so I don't know how we want to handle this. We could spend time figuring out all the `Result`s and ways they can fail, but I just wanted to be specific about that case

Reply:
Valid point, but let's split the paragraphs though for the generated doc view
```suggestion
    /// Ensures that all agent-creation commands contain valid agent fields. 
    ///
    /// Fails if a creation command is for an agent that has a field that hasn't been defined in the schema
```